meteor/scale.py

```diff
@@ -1,15 +1,222 @@
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
 import reciprocalspaceship as rs
+import scipy.optimize as opt
+
+DTYPES_TO_SCALE = (
+    rs.AnomalousDifferenceDtype,
+    rs.FriedelIntensityDtype,
+    rs.FriedelStructureFactorAmplitudeDtype,
+    rs.IntensityDtype,
+    rs.NormalizedStructureFactorAmplitudeDtype,
+    rs.StandardDeviationDtype,
+    rs.StandardDeviationFriedelIDtype,
+    rs.StandardDeviationFriedelSFDtype,
+    rs.StructureFactorAmplitudeDtype,
+)
+""" automatically scale these types when they appear in an rs.DataSet """
+
+
+ScaleParameters = Tuple[float, float, float, float, float, float, float]
+""" 7x float tuple to hold anisotropic scaling parameters """
+
+
+def _compute_anisotropic_scale_factors(
+    miller_indices: pd.Index, anisotropic_scale_parameters: ScaleParameters
+) -> np.ndarray:
+    miller_indices_as_array = np.array(list(miller_indices))
+    squared_miller_indices = np.square(miller_indices_as_array)
+
+    h_squared = squared_miller_indices[:, 0]
+    k_squared = squared_miller_indices[:, 1]
+    l_squared = squared_miller_indices[:, 2]
+
+    hk_product = miller_indices_as_array[:, 0] * miller_indices_as_array[:, 1]
+    hl_product = miller_indices_as_array[:, 0] * miller_indices_as_array[:, 2]
+    kl_product = miller_indices_as_array[:, 1] * miller_indices_as_array[:, 2]
+
+    # Anisotropic scaling term
+    exponential_argument = -(
+        h_squared * anisotropic_scale_parameters[1]
+        + k_squared * anisotropic_scale_parameters[2]
+        + l_squared * anisotropic_scale_parameters[3]
+        + 2 * hk_product * anisotropic_scale_parameters[4]
+        + 2 * hl_product * anisotropic_scale_parameters[5]
+        + 2 * kl_product * anisotropic_scale_parameters[6]
+    )
+
+    return anisotropic_scale_parameters[0] * np.exp(exponential_argument)
+
+
+def compute_scale_factors(
```

Comment:
the naming here seems in contrast to the above, with expectation that this will not be anisotropic

Reply:
do you think we should change one or the other?

personally I think this is best -- in principle, we could expand `compute_scale_factors` to be a common interface to many kinds of scaling (aniso, iso, constant) using a kwarg flag. But right now the docstring hopefully makes things clear.

Happy to hear a firm opinion if you have one.

Reply:
true that the docstring makes it clear. I like the idea of having this general function name that in the future could have optional input to kinds of scaling. So let's roll with it.